Tools/Scripts/libraries/webkitcorepy/webkitcorepy/file_lock.py

```diff
@@ -44,42 +62,60 @@ def acquired(self):
 
     def acquire(self):
         if self._descriptor:
+            raise RuntimeError('Cannot re-enter acquired FileLock')
+
+        if not self.USE_EXLOCK and not self.USE_WINDOWS and os.path.exists(self.path):
+            with open(self.path) as file:
+                pid = file.readline().strip()
+            if self.INTEGER_RE.match(pid) and not self.is_process_running(int(pid)):
+                os.unlink(self.path)
+
+        if self.USE_EXLOCK and self.timeout:
+            with Timeout(
```

Comment:
I think that this has a race. The alarm can fire after the file is locked by the kernel but before `self._descriptor` is assigned.

Reply:
Is does, yes. But this exception is always fatal in practice, so we'll end up with a dead process shortly after throwing it.